content/guides/use-case/tensorflowjs.md

```diff
@@ -0,0 +1,584 @@
+- The code for a face detection application using TensorFlow.js
+- How to create and modify the Docker assets for a TensorFlow.js application
+
+> **Acknowledgment**
+>
+> Docker would like to thank [Harsh Manvar](https://github.com/harsh4870) for
+> his contribution to the this guide.
+
+## Prerequisites
+
+* You have installed the latest version of [Docker Desktop](../../../get-docker.md).
+* You have a [Git client](https://git-scm.com/downloads). The examples in this guide use a command-line based Git client, but you can use any client.
+
+## What is TensorFlow.js?
+
+[TensorFlow.js](https://www.tensorflow.org/js) is an open-source JavaScript
+library for machine learning that enables you to train and deploy ML models in
+the browser or on Node.js. It supports creating new models from scratch or using
+pre-trained models, facilitating a wide range of ML applications directly in web
+environments. TensorFlow.js offers efficient computation, making sophisticated
+ML tasks accessible to web developers without deep ML expertise.
+
+## Why Use TensorFlow.js and Docker together?
+
+- Environment consistency and simplified deployment: Docker packages
+  TensorFlow.js applications and their dependencies into containers, ensuring
+  consistent runs across all environments and simplifying deployment.
+- Efficient development and easy scaling: Docker enhances development efficiency
+  with features like hot reloading and facilitates easy scaling of -
+  TensorFlow.js applications using orchestration tools like Kubernetes.
+- Isolation and enhanced security: Docker isolates TensorFlow.js applications in
+  secure environments, minimizing conflicts and security vulnerabilities while
+  running applications with limited permissions.
+
+
+## Get and run the sample application
+
+In a terminal, clone the sample application using the following command.
+
+```console
+$ git clone https://github.com/docker/face-detection-tensorjs
+```
+
+After cloning the application, you'll notice the application has a `Dockerfile`
+and a `compose.yaml` file. These Docker assets let you build and run the
+application locally with nothing more than Docker.
+
+To run the application, run the following command in a terminal
+inside the `face-detection-tensorjs` directory.
+
+```console
+$ docker compose up --build
+```
+
+The command builds the application into an image and runs it. Depending on your
+network connection, it can take several minutes to download the dependencies the
+first time you run the command.
+
+You'll see the following output when the application starts.
+
+```console
+[+] Running 1/0
+ ✔ Container face-detection-tensorjs-server-1  Created                                                                             0.0s 
+Attaching to server-1
+server-1  | yarn run v1.22.19
+server-1  | $ cross-env BUILD_PATH='./static' NODE_ENV=development parcel index.html --no-hmr
+server-1  | Server running at http://localhost:1234 
+server-1  | ✨  Built in 3.87s.
```

Comment:
This step took way longer for me on an ARM mac on the first run.

```
server-1  | ✨  Built in 19.35s.
```

Subsequent builds were faster

```
server-1  | ✨  Built in 1.89s.
```

Not sure if that's just me or general. If it's typically around 20s for the first startup we could show that in this output I guess